cellarium/ml/data/distributed_anndata.py

```diff
@@ -212,43 +213,44 @@ def __init__(
                 indices_strict=indices_strict,
             )
 
-    def __getitem__(self, index: Index) -> DistributedAnnDataCollectionView:
+    def __getitem__(self, index: Index) -> AnnData:
         """
-        Return a distributed view of the collection of anndatas.
+        Materialize and gather anndata files at given indices from the list of lazy anndatas.
 
         :class:`LazyAnnData` instances corresponding to cells in the index are materialized.
         """
         oidx, vidx = _normalize_indices(index, self.obs_names, self.var_names)
-        resolved_idx = self._resolve_idx(oidx, vidx)
-        adatas_indices = [i for i, e in enumerate(resolved_idx[0]) if e is not None]
-        # TODO: materialize at the last moment?
-        self.materialize(adatas_indices)
-
-        return DistributedAnnDataCollectionView(self, self.convert, resolved_idx)
+        adatas_oidx, oidx, vidx, reverse = self._resolve_idx(oidx, vidx)
+        adatas = self.materialize(adatas_oidx, vidx)
+        adata = concat(adatas, merge="same")
+        adata = adata if reverse is None else adata[reverse]
+        # make sure that categorical dtypes are preserved
+        adata.obs = adata.obs.astype(self.schema.attr_values["obs"].dtypes)
+        return adata
```

Comment:
Now this returns a single gathered anndata object instead of `DistributedAnnDataCollectionView` thus bypassing the logic from `AnnCollection`. `DistributedAnnDataCollectionView` is only useful if we slice lazily multiple times which we never do.

Another difference here is that now we gather and concatenate entire anndata objects at once. With `DistributedAnnDataCollectionView` gathering/concatenating happens for each attribute that is accessed (`X`, `obs`, etc) which in edge cases can lead to loading the same anndata file multiple times when multiple attributes are accessed.